### ISSUE TITLE
ignore ROSE_VERSION/CYLC_VERSION in rose-suite.conf

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-rose#11]
#### tldr

Exclude any `ROSE_VERSION` and `CYLC_VERSION` environment variables set in the `rose-suite.conf` from being exported into the environment, messing with templatvars or whatever.

#### context

In Rose2019 there was some level of support for the `[env]ROSE_VERSION` and `[env]CYLC_VERSION` variables.

If `ROSE_VERSION` was set and did not match `rose version` then rose exited with an error.

If `CYLC_VERSION` was set Rose exported it to `cylc` executable calls (to pick up the correct version via the wrapper script).

Now that cylc-rose is a plugin and Rose2/Cylc8 are installed in the same Python environment this system is meaningless and no longer functions.

However, many Rose2019 suites will have these variables defined, they have the potential to mess with the "true" `*_VERSION` variables and cause problems.

**Comment by wxtim** [comment]
On examination current behavior sets `ROSE_VERSION` is already hard-set to `from metomi.rose import __version__ as ROSE_VERSION` - is this ok, or should I add a warning that this has happened?

**Comment by oliver-sanders** [comment]
`ROSE_VERSION`

Hardcoding the `ROSE_VERSION` to `metomi.rose.__version__` is absolutely correct :+1:.

`CYLC_VERSION`

Cylc handles this itself so just ignore.

**Comment by oliver-sanders** [comment]
(raise warnings if either are set because we need to stop people using these settings now that they are deprecated, they can only contribute to confusion)

**Comment by oliver-sanders** [comment]
From team meeting a few weeks back.

We would like to be rid of these as they cause a lot of needless bother, the only effective way to do so is to ignore them with a warning.

* Cylc 8 should provide some passive protection against **restarting** workflows with incorrect versions.
* The CLI is now prominently GraphQL so Cylc versions should be fairly inter-compatible.

What remains is to determine what use case remains for specifying the Cylc/Rose version on startup.

**Comment by oliver-sanders** [comment]
I've just run a quick scan of Rose 2019 suites, there are still quite a lot (thousands) still using a Cylc 5 workaround:

```ini
        init-script = """    
           export CYLC_VERSION={{CYLC_VERSION}}                                                                                                                                                                    
           export ROSE_VERSION={{ROSE_VERSION}}    
        """    
```

We've tried to get these removed, however, there are still a lot kicking around.

So I think we still need `ROSE_VERSION` and `CYLC_VERSION` to be provided as env & Jinja2 vars, however, we would like them to be set from the actual Rose/Cylc versions rather than from whatever was set in the `rose-suite.conf`.

So I think:

* Continue to provide these vars.
* But set them using the version obtained via Python import.
* And raise a warning if they are set in the `rose-suite.conf` file.
* (later) Consider checking Cylc `*script` strings for this pattern and raising a warning.

**Comment by oliver-sanders** [comment]
Decided to:

* Continue to provide these vars.
* But set them using the version obtained via Python import.
* And raise a warning if they are set in the rose-suite.conf file.

**Comment by oliver-sanders** [comment]
Closed by #86 (which is now released so removing the incorrect milestone)
